src/plugins/cachefilter/cachefilter.c

```diff
@@ -42,7 +42,7 @@ int elektraCachefilterGet (Plugin * handle, KeySet * returned, Key * parentKey)
 	void * cache = elektraPluginGetData (handle);
 	if (cache == NULL)
 	{
-		toBeCached = ksNew (ksGetSize (returned), KS_END);
+		toBeCached = ksNew (0, KS_END);
```

Comment:
No I didn't but this change is also not crucial.

The code of `ksAppend`, which is used later here, does suggest that it uses only one resize in advance.

I wanted to let the core decide about the allocation but if you're saying this is bad I don't mind undoing this. The objective was to get rid of the memleaks and move forward.